packages/alfa-rules/test/sia-r82/rule.spec.tsx

```diff
@@ -0,0 +1,146 @@
+import { jsx } from "@siteimprove/alfa-dom/jsx";
+import { test } from "@siteimprove/alfa-test";
+
+import { Option } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+
+import { Document, Element } from "@siteimprove/alfa-dom";
+
+import R82, { Outcomes } from "../../src/sia-r82/rule";
+
+import { evaluate } from "../common/evaluate";
+import { oracle } from "../common/oracle";
+import { passed, failed } from "../common/outcome";
+
+const { isElement, hasName } = Element;
+const { and } = Predicate;
+
+const document = Document.of((self) => [
+  Element.fromElement(
+    <form>
+      <label>
+        Input
+        <input type="text"></input>
+      </label>
+      <span>Visible error</span>
+      <span hidden>Invisible error</span>
+      <span aria-hidden="true">Ignored error</span>
+    </form>,
+    Option.of(self)
+  ),
+]);
+
+const input = document
+  .descendants()
+  .find(and(isElement, hasName("input")))
+  .get();
+
+const [error, invisibleError, ignoredError] = document
```

Comment:
```suggestion
const [visibleError, invisibleError, ignoredError] = document
```
would be more consistent naming and help parsing the examples further down.